### ISSUE TITLE
[OpenVSX] Use improved `/api/-/query` endpoint

### ISSUE THREAD

**Issue by amvanbaren** [eclipse-theia/theia#11291]
Try out the `/api/v5/-/query` endpoint as described in https://github.com/eclipse/openvsx/pull/454#issuecomment-1138602969

### Feature Description:
- Replace calls to the `/api/-/query` endpoint with calls to the `/api/v1/-/query` endpoint. (`/api/v5/-/query` will be renamed to `/api/v1/-/query`)
- Try to use the `includeAllVersions=false` combined with `extensionVersion=${version}` parameters as much as possible, so that responses can be cached.
- `includeAllVersions=true` returns data for all extension versions. The `url` field contains the link for that extension version (instead of the `allVersions` map that was included for every extension version).
- `includeAllVersions=links` returns data for the latest or a specific version. This extension version object does contain the `allVersions` map for further browsing.

**Comment by ccorn** [comment]
You mean, this is live on `https://open-vsx.org/api/v1/-/query`? It does not seem to be:

    $ curl -LsS --compressed 'https://open-vsx.org/api/v1/-/query?extensionId=vscode.bat'
    {"error":"Extension not found: v1.- version query"}

Same with `v5`.

The gitpod instance for [eclipse/open-vsx#454](https://github.com/eclipse/openvsx/pull/454) should still work (with `v5`) however.

**Comment by msujew** [comment]
@ccorn Since https://github.com/eclipse/openvsx/pull/454 isn't merged yet, it's not live on `open-vsx.org`.

@amvanbaren I'd be happy to make the change on Theia's side once the PR is merged and the API endpoint is available on `open-vsx.org`. When do you plan on merging it?

**Comment by amvanbaren** [comment]
> When do you plan on merging it?

I'm still waiting on feedback for a fix of the current release: https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/1377
This is blocking all other PRs, including https://github.com/eclipse/openvsx/pull/454. 

I'll refactor https://github.com/eclipse/openvsx/pull/454 to only the final endpoint (we did 4-5 iterations of the endpoint), so that it's ready to be merged.
For now this issue can be put on hold, I'll add a comment once https://github.com/eclipse/openvsx/pull/454 is live on `open-vsx.org`.

**Comment by amvanbaren** [comment]
@msujew The `/api/v2/-/query` endpoint is live on open-vsx.org. It's documented here: https://open-vsx.org/swagger-ui/index.html#/registry-api/getQueryV2.

**Comment by amvanbaren** [comment]
> Yes, the meaning of identifiers with allVersions in them is heavily overloaded (i.e. context-dependent) here.

@ccorn, @msujew Would it be a better idea to add an `allVersionsUrl` field that points to a `/api/{namespace}/{extension}/versions` endpoint? I think that would be a less context dependent solution. The endpoint returns 10 - 50 versions at a time. A response for `/api/foo/bar/versions?offset=20&size=10` would look like this:
```
{
  "versions": {
    "1.0.0": "https://open-vsx.org/api/foo/bar/1.0.0",
    "0.8.0": "https://open-vsx.org/api/foo/bar/0.8.0"
    ...
  },
  "offset": 20,
  "totalSize": "52"
}
```

The `allVersionsUrl` field can be added to the current `/api/-/query` response alongside the `allVersions` field. However, the `allVersions` field will no longer contain all versions, just the last 200. Over time the `allVersions` field will contain less and less versions, until at some point it only contains the last 5 - 10 versions or maybe even 0. `allVersions` will be marked as deprecated in the docs to encourage the use of `allVersionsUrl` instead.
The `/api/v2/-/query` endpoint will no longer have the `allVersions` field, just the `allVersionsUrl`. At some point `/api/-/query` requests will be redirected to the `/api/v2/-/query` endpoint.

What do you think?
